translate-toolkit: Init at 2.2.5

phile314 opened the pull request:
###### Motivation for this change

Add the translation-toolkit tools:
http://toolkit.translatehouse.org/

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

phile314 committed:
translate-toolkit: Init at 2.2.5

Mic92 commented:
Is this supposed to be used as a library or a set of command-line libraries.

phile314 commented:
In the end it's both as it has a lot of executable tools ( http://docs.translatehouse.org/projects/translate-toolkit/en/latest/commands/index.html ) but also an API ( http://docs.translatehouse.org/projects/translate-toolkit/en/latest/api/index.html ). I am only using the tools and I think those are more important/popular. Is there an easy way to make it both a normal library package and a set of command-line tools in nixpkgs?

Mic92 commented on pkgs/tools/misc/translate-toolkit/default.nix:
```diff
@@ -0,0 +1,31 @@
+{ stdenv, fetchFromGitHub, python3Packages, makeWrapper }:
+
+python3Packages.buildPythonApplication rec{
```
We don't store python applications here, as we we build `pythonPackages` against multiple python versions.
Please change to `buildPythonLibrary` here.

Mic92 commented on pkgs/tools/misc/translate-toolkit/default.nix:
```diff
@@ -0,0 +1,31 @@
+  propagatedBuildInputs = with python3Packages; [
+    six diff-match-patch lxml python-Levenshtein chardet pycountry
+  ];
+  checkInputs = with python3Packages; [ pytest ];
```
This would make building against `python2` impossible. Please move all libraries into the header and remove the `python3Packages`.

```
{ stdenv, fetchFromGitHub, python3Packages, makeWrapper, six, diff-match-patch, ... }:
```

Mic92 commented:
Packages which can be used as both stay in `python-packages.nix`. However we sometimes add a reference from `all-packages.nix`:

```
translate-toolkit = python3Packages.translate-tookit;
```

phile314 commented:
The translate-toolkit is now a normal python library. I added a reference in ``all-packages.nix`` as suggested as it also contains some useful executables. Also rebased & squashed on current master.

jtojnar commented:
Could you actually move the package to `pkgs/development/python-modules/translate-toolkit` and then use `callPackage` inside `python-packages.nix`?

FRidh commented on pkgs/top-level/python-packages.nix:
```diff
@@ -20710,6 +20710,36 @@ in {
 
   transaction = callPackage ../development/python-modules/transaction { };
 
+  translate-toolkit = buildPythonPackage rec {
```
We are moving Python expressions out of `pkgs/top-level/python-packages.nix` into `pkgs/development/python-modules/<module>/default.nix`. 

Please move the expression there, and call it from `pkgs/top-level/python-packages.nix` using `callPackage ../development/python-modules/<package> { };`.

Ekleog commented:
(triage) @phile314, are you still willing to push this forward?

jtojnar commented:
I am actually working on this (packaging weblate).

phile314 commented:
Feel free to take over or close this, I no longer have the time to push this.

FRidh commented:
@jtojnar please open a new PR with your version.

jtojnar commented:
See https://github.com/NixOS/nixpkgs/pull/48726
